src/d/actor/d_a_fganon.cpp

```diff
@@ -5,195 +5,2885 @@
+                        }
+                        i_this->m388++;
+                        fopAcM_monsSeStart(i_this, JA_SE_CV_PG_REFLECT_EBALL, 0);
+                        mDoAud_seStart(JA_SE_CM_PG_SWING_S, &i_this->eyePos, 0, dComIfGp_getReverb(fopAcM_GetRoomNo(i_this)));
+                        i_this->m687++;
+                        if ((i_this->m688 >= 7) || ((i_this->m687 >= 4) && (cM_rndF(1.0f) < 0.3f))) {
+                            i_this->m689 = 1;
+                        }
+                    }
+                    else if (i_this->m2BC == 2) {
+                        i_this->m386 = 2;
+                        i_this->m388 = 0;
+                        return;
+                    }
+                }
+            }
+            break;
+        }
+        case 4: {
+            if (((s32)i_this->mpMorf->getFrame() >= 5) && ((s32)i_this->mpMorf->getFrame() <= 15)) {
+                i_this->m686 = 1;
+            }
+            if (i_this->mpMorf->isStop() || ((i_this->m2BC >= 1) && (i_this->mpMorf->getFrame() >= 20.0f)))
+            {
+                anm_init(i_this, 0x16, 10.0f, 2, 1.0f, -1);
+                i_this->m388 = 3;
+            }
+            // Fall-through
+        }
+    }
+    cLib_addCalc0(&i_this->speedF, 1.0f, REG0_F(14) + 1.0f);
+    pos_move(i_this, 0);
+    fly_se_set(i_this);
+    if ((i_this->m388 >= 2) && (i_this->m408 == 0)) {
+      i_this->m386 = 5;
+      i_this->m388 = 0;
+    }
+    if ((i_this->m388 == 3) || (i_this->m388 == 4)) {
+        mDoAud_monsSeStart(JA_SE_OBJ_PG_EBALL_FLY_S, &i_this->m3E0, 100, dComIfGp_getReverb(fopAcM_GetRoomNo(i_this)));
+    }
+    return;
 }
 
 /* 00001BD0-000020B8       .text spinattack__FP12fganon_class */
-void spinattack(fganon_class*) {
-    /* Nonmatching */
+void spinattack(fganon_class* i_this) {
+    fopAc_ac_c* a_this = (fopAc_ac_c*)i_this;
+    fopAc_ac_c* player = (fopAc_ac_c*)dComIfGp_getPlayer(0);
+    dBgS_LinChk linChk;
+    int mFrame = i_this->mpMorf->mFrameCtrl.getFrame();
+    switch(i_this->m388) {
+        case 0: {
+            i_this->shape_angle.y = player->shape_angle.y;
+            i_this->m388 = 1;
+            // Fall-through
+        }
+        case 1: {
+            mDoMtx_YrotS(*calc_mtx, i_this->shape_angle.y);
+            cXyz posVec(0.0f, l_HIO.m2C + 100.0f, l_HIO.m30);
+            cXyz transformedPos;
+            MtxPosition(&posVec, &transformedPos);
+            i_this->current.pos = player->current.pos + transformedPos;
+            linChk.Set(&player->eyePos, &i_this->current.pos, a_this);
+            if (!dComIfG_Bgsp()->LineCross(&linChk)) {
+                i_this->current.pos.y -= 100.0f;
+                anm_init(i_this, 0x16, 1.0f, 2, 1.0f, -1);
+                i_this->m388 = 2;
+                i_this->m3A4[0] = 30;
+                deru_brk(i_this);
+                // Fall-through
+            }
+            else {
+                i_this->shape_angle.y = cM_rndFX(32768.0f);
```

Comment:
```suggestion
                i_this->shape_angle.y = cM_rndFX(0x8000);
```
Even though the arg is a float the fact that it's an angle means they likely wrote it as hex